pennylane/plugins/expt_tensornet_tf.py

```diff
@@ -0,0 +1,360 @@
+
+        self.eng = None
+        self.analytic = True
+        self._nodes = []
+        self._edges = []
+        zero_state = np.zeros([2] * wires)
+        zero_state[tuple([0] * wires)] = 1.0
+        self._zero_state = tf.constant(zero_state, dtype=tf.complex128)
+        # TODO: since this state is separable, can be more intelligent about not making a dense matrix
+        self._state_node = self._add_node(
+            self._zero_state, wires=tuple(w for w in range(wires)), name="AllZeroState"
+        )
+        self._free_edges = self._state_node.edges[:]  # we need this list to be distinct from self._state_node.edges
+
+    def _get_operator_matrix(self, operation, par):
+        """Get the operator matrix for a given operation or observable.
+
+        Args:
+          operation    (str): name of the operation/observable
+          par (tuple[float]): parameter values
+        Returns:
+          array: matrix representation.
+        """
+        A = {**self._operation_map, **self._observable_map}[operation]
+        if not callable(A):
+            return tf.constant(A, dtype=tf.complex128)
+        return tf.convert_to_tensor(A(*par), dtype=tf.complex128)
+
+    def apply(self, operation, wires, par):
+        if operation == "QubitStateVector":
+            state = tf.constant(par[0], dtype=np.complex128)
+            if state.ndim == 1 and state.shape[0] == 2 ** self.num_wires:
+                self._state_node.tensor = tf.reshape(state, [2] * self.num_wires)
+            else:
+                raise ValueError("State vector must be of length 2**wires.")
+            if wires is not None and wires != [] and list(wires) != list(range(self.num_wires)):
+                raise ValueError(
+                    "The expt.tensornet plugin can apply QubitStateVector only to all of the {} wires.".format(
+                        self.num_wires
+                    )
+                )
+            return
+        if operation == "BasisState":
+            n = len(par[0])
+            if n == 0 or n > self.num_wires or not set(par[0]).issubset({0, 1}):
+                raise ValueError(
+                    "BasisState parameter must be an array of 0 or 1 integers of length at most {}.".format(
+                        self.num_wires
+                    )
+                )
+            if wires is not None and wires != [] and list(wires) != list(range(self.num_wires)):
+                raise ValueError(
+                    "The expt.tensornet plugin can apply BasisState only to all of the {} wires.".format(
+                        self.num_wires
+                    )
+                )
+            stat_node = np.zeros(tuple([2] * len(wires)))
+            stat_node[tuple(par[0])] = 1
+            self._state_node.tensor = tf.constant(stat_node, dtype=tf.complex128)
+            return
+
+        A = self._get_operator_matrix(operation, par)
+        num_mult_idxs = len(wires)
+        A = tf.reshape(A, [2] * num_mult_idxs * 2)
```

Comment:
Only the parametrized matrices are defined in this file :slightly_smiling_face:  The non-parametrized ones will still have in the incorrect shape. As a result, I would rather set the shape once (here), rather than setting the shape everywhere.